extensions/amp-bind/0.1/bind-expression.js

```diff
@@ -35,6 +35,46 @@ const BUILT_IN_FUNCTIONS = 'built-in-functions';
  * @const @private {!Object<string, !Object<string, Function>>}
  */
 const FUNCTION_WHITELIST = (function() {
+
+  /**
+   * Custom wrappers for mutating array methods so that we can offer the
+   * same functionality without allowing mutation on variables in bind scope.
+   */
+  const BindArrays = {
+    /**
+     * Similar to Array.prototype.splice, except it returns a copy of the
+     * passed-in array with the desired modifications.
+     * @param {!Array} array
+     * @param {number=} start
+     * @param {number=} end
+     * @param {...?} items
+     */
+    /*eslint "no-unused-vars": 0*/
+    'arraySplice': function(array, start, end, items) {
+      const copy = array.slice();
```

Comment:
Right now we're doing some checking that the caller has the method, AND that the whitelist has the method, AND the two methods are the same. We could weaken that check or add the methods to Array.prototype, but I don't think neither of those solutions is viable. (I'm assuming the check on line 233 is there for security reasons). 

I also think making these built-ins makes it obvious that these functions are distinct from the methods on Array. The type check shouldn't be too unwieldy.